File: python/src/wdl_model/model/base/wdl_node.py
"""Base node type for the Python WDL object model.

The library models a WDL document as a tree of nodes representing top-level definitions,
statements, sections, expressions, and types. See the WDL 1.3 specification sections "WDL
Documents", "Task Definition", and "Workflow Definition" for the source-language concepts.
"""


class WdlNode:
    """Marker base class matching Java WdlNode interface."""

File: java/src/main/java/com/myriad/wdl/model/base/WdlNode.java
package com.myriad.wdl.model.base;

/**
 * Marker interface for every node in the Java WDL object model.
 *
 * <p>The library models a WDL document as a tree of nodes representing top-level definitions,
 * statements, sections, expressions, and types. See the WDL 1.3 specification sections "WDL
 * Documents", "Task Definition", and "Workflow Definition" for the grammar concepts represented
 * by these nodes.
 */
public interface WdlNode {}
